### PR TITLE
5 question and question vote for user

### PR DESCRIPTION
**Pull request by PhanVietTrung**
#### Status

READY 

#### What's this PR do?
CRUD Question, Question vote

### CODE REVIEW

File: lib/liulo_web/controllers/question_controller.ex [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+      questions
+      |> Enum.map(fn(%Question{:id => id}) ->
+        question_votes
+        |> Enum.filter(
+          fn(%QuestionVote{:question_id => question_id}) -> question_id == id end)
+        |> Enum.count() > 0
+      end)
+
+
+    # results = Enum.zip(is_voteds, questions)
```
Comment (thminhVN) [inline lib/liulo_web/controllers/question_controller.ex]: Remove unused code

File: lib/liulo_web/controllers/question_controller.ex [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+    # IO.inspect "------------------"
+    # IO.inspect results
+    render(conn, "is_voted.json", questions: questions, is_votes: is_voteds)
```
Comment (thminhVN) [inline lib/liulo_web/controllers/question_controller.ex]: Please make parameter name consistent

Replace `is_votes` -> `is_voted`

File: lib/liulo_web/controllers/question_controller.ex [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+  end
+
+  def create(conn, %{"topic_id" => topic_id, "question" => question_params}) do
+    # topic = Liulo.Repo.get_by!(Topic, id: topic_id)
```
Comment (thminhVN) [inline lib/liulo_web/controllers/question_controller.ex]: Remove unused code

File: lib/liulo_web/controllers/question_controller.ex [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+    # owner =  Liulo.Guardian.Plug.current_resource(conn)
+    # {:ok, question} = Events.create_question(owner, topic, question_params)
+    with %Topic{} = topic <- Liulo.Repo.get_by!(Topic, id: topic_id),
+          %User{} = owner <-  Liulo.Guardian.Plug.current_resource(conn),
+        {:ok, %Question{} = question} <- Events.create_question(owner, topic, question_params) do
+      IO.inspect question
```
Comment (thminhVN) [inline lib/liulo_web/controllers/question_controller.ex]: Remove this

File: lib/liulo_web/controllers/question_vote_controller.ex [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+
+  action_fallback LiuloWeb.FallbackController
+
+  def create(conn, %{ "question_id" => question_id}) do
+    upvote_user = Liulo.Guardian.Plug.current_resource(conn)
+    question = Events.get_question!(question_id)
+    existed = Repo.get_by(QuestionVote, [user_id: upvote_user.id , question_id: question.id]) != nil
+    if existed == false do
+      with {:ok, %{question_vote: question_vote}} <- Events.create_question_vote(question, upvote_user) do
+        IO.inspect "--------------------------"
```
Comment (thminhVN) [inline lib/liulo_web/controllers/question_vote_controller.ex]: remove this

File: test/liulo_web/controllers/question_controller_test.exs [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+  describe "create question" do
+    test "renders question when data is valid", %{conn: conn} do
+      topic = insert(:topic)
+
+      post_conn = post conn, topic_question_path(conn, :create, topic.id), question: @create_attrs
+      assert json_response(post_conn, 201)["data"]
+
+    end
+
+    # test "renders errors when data is invalid", %{conn: conn} do
```
Comment (thminhVN) [inline test/liulo_web/controllers/question_controller_test.exs]: Remove unused tests

File: lib/liulo_web/controllers/question_controller.ex (hunk 2 above)
Comment (PhanVietTrung) [reply lib/liulo_web/controllers/question_controller.ex]: this is array of votes

File: lib/liulo/events/events.ex [hunk 7]
```diff
@@ -204,4 +205,154 @@ defmodule Liulo.Events do
...
+
+  ## Examples
+
+      iex> list_question()
+      [%Question{}, ...]
+
+  """
+  def list_question_by_topic(%Topic{} = topic) do
+    topic = topic |> Repo.preload(:questions)
+    topic.questions |> Enum.sort_by(fn(p) -> p.vote_count end)
```
Comment (thminhVN) [inline lib/liulo/events/events.ex]: Should be sorted in query

File: lib/liulo/events/events.ex [hunk 8]
```diff
@@ -204,4 +205,154 @@ defmodule Liulo.Events do
...
+      {:error, %Ecto.Changeset{}}
+
+  """
+  def delete_question_vote(question_id, user_id) do
+    query = from qv in QuestionVote, where: qv.user_id == ^user_id and qv.question_id == ^question_id
+    Repo.delete_all(query)
+  end
+  defp update_number_of_vote_for_question(question) do
+    query = from qv in QuestionVote, where: qv.question_id == ^question.id
+    with count <- Repo.all(query) |> Enum.count do
```
Comment (thminhVN) [inline lib/liulo/events/events.ex]: Please use `Repo.aggregate/4` to count data

File: lib/liulo_web/controllers/question_controller.ex [hunk 9]
```diff
@@ -0,0 +1,74 @@
...
+  def question_by_topic(conn, %{"topic_id" => id}) do
+    owner_id = Liulo.Guardian.Plug.current_resource(conn).id
+
+    topic = Events.get_topic!(id)
+    questions = Events.list_question_by_topic(topic) |> Enum.sort_by(fn(p) -> p.vote_count end)
+
+    query = from qv in QuestionVote, where: qv.user_id == ^owner_id
+    question_votes = Repo.all(query)
+
+    is_voteds =
```
Comment (thminhVN) [inline lib/liulo_web/controllers/question_controller.ex]: should be `is_voted`

**Pull request merged**